README.md

```diff
@@ -15,23 +15,56 @@ The other clones a [TNA fork](https://github.com/nationalarchives/keycloak-sms-a
 
-### import-realm shell script
+## import_env_realm.py script
 
-The import-realm.sh runs scripts to update the Keycloak json configuration file (which is held in a private repository).
+The import_env_realm.py runs scripts to update the Keycloak json configuration files (which are held in a private repository) with TDR environment specific properties and combines the individual realm json into a single json configuration file for import into Keycloak at start up.
 
-Primarily it injects secret values which cannot be safely stored in the configuration file.
+## update_env_client_configuration.py script
 
-### Configuration file
+The update_env_client_configuration.py script provides functions for updating Keycloak realm client json. 
+
+Primarily it:
+ * injects secret values which cannot be safely stored in the configuration file.
+ * updates specific json elements based on the environment properties json file provided.
+ 
+## tdr-entrypoint.sh script
+
+The tdr-entrypoint.sh script provides the entry point on start up of the Keycloak docker container.
+
+It has specific TDR commands around realm import, and is copied from the default entrypoint script provided in the keycloak image (/opt/jboss/tools/docker-entrypoint.sh)
+
+## Configuration file
 The standalone-ha.xml is mostly the standard configuration for keycloak with a few changes to get it to work with the load balancer. Some of these are discussed in the keycloak [documentation](https://www.keycloak.org/docs/latest/server_installation/#_setting-up-a-load-balancer-or-proxy)
 
-### Updating Keycloak Configuration json
+## Updating Keycloak Configuration json
 
 To update Keycloak with, for example, a new client:
-1. Update the Keycloak json configuration file. See README for the tdr-configuration private repository on how to do this.
+1. Update the relevant Keycloak json configuration file (tdr-realm-export.json or master-realm-export.json). See README for the tdr-configuration private repository on how to do this.
 2. If the change to Keycloak makes use of a new secret value, for example a new client secret:
   * Add the new secret value to the parameter store using Terraform: https://github.com/nationalarchives/tdr-terraform-environments
     
     This ensures that the secret value is stored securely and is not exposed in the code.
   
-  * Update the import-realm.sh script to replace the placeholder secret value in the realm-export.json configuration file, with the new secret value set in the Terraform.
+  * Update the update_env_client_configuration.py script to replace the placeholder secret value in the relevant realm json configuration file, with the new secret value set in the Terraform.
 3. Run the Jenkins build
+
+## Running Locally
+
+To run, build and test locally:
+1. Copy the master-realm-export.json and tdr-realm-export.json from the tdr-configuration repository into the tdr-auth-server directory
```

Comment:
Just wondering if it would be easier to do a git clone, like we do for the Terraform modules project inside the other Terraform projects?

Reply:
Possibly, but then running locally would require changes to the paths to point to the realm export json files within the clone repo in the python scripts.